src/arkode/arkode_mristep.c

```diff
@@ -1058,22 +1053,23 @@ int mriStep_Init(void* arkode_mem, int init_type)
     }
 
     /* Retrieve/store method and embedding orders now that tables are finalized */
+    delta_stages = step_mem->MRIC->stages - step_mem->stages;
     step_mem->stages = step_mem->MRIC->stages;
     step_mem->q      = step_mem->MRIC->q;
     step_mem->p      = step_mem->MRIC->p;
 
     /* allocate/fill derived quantities from MRIC structure */
 
     /* stage map */
-    if (step_mem->stage_map)
+    step_mem->stage_map = realloc(step_mem->stage_map,
+                                  step_mem->stages * sizeof(*step_mem->stage_map));
```

Comment:
I'm okay with changing to `realloc`, but (a) please cast the result to the correct type (`int*`), and (b) `memset` the new memory to all zeros to retain `calloc`'s initialization feature.

Reply:
The `calloc` is not necessary, since the `stage_map` will be filled in by `mriStepCoupling_GetStageMap`. I'll concede this could be a bit fragile if that function changes, but it's also easier to detect issues using valgrind in the current form. I don't have a strong preference, so I'll add the `memset` if you think it's necessary.

Reply:
`memset` is not need for any of the `realloc`s introduced in this PR. I checked each usage and valgrind doesn't report any issues. Certainly in other parts of the code outside this PR, we may need the `memset` if we switch to `realloc`. For this PR, are you ok with leaving out the `memset`?

Reply:
I'm not sure how much of a benefit there is from `realloc` in this case since we'll overwrite the existing data in the array. If `realloc` can grow or shrink the allocation then we save the `free` call and if not then there is an unnecessary copy.

Reply:
I recall that we've had problems in the past with using `malloc` without `memset`, but only on some compilers and platforms (I seem to recall Windows and MS Visual Studio being particularly problematic), which is why we're being careful to use `calloc` nowadays.  It's been my experience that GCC on Linux does what we "expect," so I'm not surprised that valgrind didn't report any issues.

Given that this purpose of this PR is to add things in a completely different location (new MRI tables), I would prefer that it not fiddle around with *anything* in this function.  However, your overall `calloc` -> `realloc` approach is reasonable, so I'm fine with including it in this PR, as long as we can guarantee that these changes will work on all platforms/compilers.

Reply:
Good point David, I completely forgot about `realloc` copying the old data. I'll revert back to `free`+`calloc` while keeping the fix for `cvals` and `Xvecs` reallocation. That will be consistent and safe and we can tackle the optimal allocation question later.